File: oddqueries.py
t = int(input())
for i in range(t):
    n,q = map(int,input().split())
    a = list(map(int,input().split()))
    for i in range(q):
        l,r,k = map(int,input().split())
        b=[]
        if(l-1 == 0):
            b=a[(l-1):r]
            c = a[r:]
            d =len(b) * k
            if((d+sum(c))%2!=0):
                print("YES")
            else:
                print("NO")
        else:
            b1 = a[(l-1):r]            
            c1 = a[:(l-1)]
            c2 = a[r:]
            d = len(b1)*k
            if((sum(c1)+sum(c2)+d)%2!=0):
                print("YES")
            else:
                print("NO")     
                 
        

File: oddqueries.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin>>t;
    while(t--){
        int n,q;
        cin>>n>>q;
        long long a[n];
         long long sum=0;
        for(int i=0;i<n;i++){
            cin>>a[i];
            sum+=a[i];
        }
       
        for(int i=1;i<n;i++){
             a[i]+=a[i-1];
        }
        while(q--){
            long long k;
            int l,r;
            cin>>l>>r>>k;
            int sum;
            if(l-2 >=0){
                long long sum1 = a[r-1] - a[l-2];
                 sum = a[n-1] - sum1;
            }else{
                 sum = a[n-1] - a[r-1];
            }
            int sol = sum + ((r - l + 1) * k);
            if((sol)%2!=0){
                cout<<"YES"<<endl;
            }
            else{
                cout<<"NO"<<endl;
            }
        }

    }
    return 0;
}